saleor/graphql/order/types.py

```diff
@@ -1930,21 +1930,26 @@ def resolve_status_display(root: models.Order, _info):
     def resolve_can_finalize(root: models.Order, info):
         if root.status == OrderStatus.DRAFT:
 
-            def _validate_draft_order(manager):
+            @allow_writer_in_context(info.context)
```

Comment:
Why do we need a writer to validate the draft order?

Reply:
Some fields that are needed to validate the draft order are fetched ad-hoc, without dataloaders. So that's why for mutations, it has to be allowed in this way. The proper solution is to add dataloaders and introduce a similar structure to `CheckoutInfo` but for orders. I can open a separate task for this, as this requires quite a lot refactoring to be done right.